uritemplate/validator.py

```diff
@@ -0,0 +1,58 @@
+"""Module containing all the validation logic for uritemplate."""
+from uritemplate import exceptions as exc
+
+
+class Validator(object):
+    """Provide configurable validation of URITemplate objects.
+
+    .. versionadded:: 3.1.0
+
+    .. code-block::
+
+        from uritemplate import URITemplate, Validator
+
+
+    """
+
+    def __init__(self):
+        """Initialize our validator."""
+        self.enforcing_unbalanced_braces = True
```

Comment:
We don't document it, so instead of adding an underscore, I just left it public. If people want to try to mutate it, then oh well. Turning this into a `property` with a setter will mean that people might try to assign weird values and it breaks chaining.